store/sqlstore/upgrade.go

```diff
@@ -1310,7 +1310,34 @@ func upgradeDatabaseToVersion600(sqlStore *SqlStore) {
 	sqlStore.AlterColumnTypeIfExists("Users", "Props", "JSON", "jsonb")
 	sqlStore.AlterColumnTypeIfExists("Users", "NotifyProps", "JSON", "jsonb")
 	sqlStore.AlterColumnTypeIfExists("Users", "Timezone", "JSON", "jsonb")
-
+	fixCRTChannelCountsAndUnreadsUpdateColumn(sqlStore)
 	// saveSchemaVersion(sqlStore, Version600)
 	// }
 }
+
+func fixCRTChannelCountsAndUnreadsUpdateColumn(sqlStore *SqlStore) {
+	created := sqlStore.CreateColumnIfNotExists("ChannelMembers", "CRTFixDone", "tinyint(1)", "boolean", "1")
+	if !created {
+		return
+	}
+
+	setFalseForUnreadChannels := `
+		UPDATE ChannelMembers
+		SET CRTFixDone=false
+		FROM Channels
+		WHERE Channels.Id = ChannelMembers.ChannelId AND Channels.TotalMsgCountRoot > ChannelMembers.MsgCountRoot;
+	`
+
+	if sqlStore.DriverName() == model.DatabaseDriverMysql {
+		setFalseForUnreadChannels = `
+			UPDATE ChannelMembers
+			INNER JOIN Channels on Channels.Id = ChannelMembers.ChannelId
+			SET CRTFixDone=false
+			WHERE Channels.TotalMsgCountRoot > ChannelMembers.MsgCountRoot;
+		`
+	}
```

Comment:
We set unread channelmembers as not fixed. Then in the job we check each channelmember and mark as read and/or fixed. On my test DB (docker mysql 5.7.12) containing 4.2 million channel members, this takes about 7.5 minutes, 1.5M of the ChannelMembers were unread.